lib/spack/docs/module_file_support.rst

```diff
@@ -0,0 +1,664 @@
+Suppose you need to avoid having ``CPATH`` and ``LIBRARY_PATH``
+modified by your ``dotkit`` modules:
+
+.. code-block:: yaml
+
+   modules:
+     dotkit:
+       all:
+         filter:
+           # Exclude changes to any of these variables
+           environment_blacklist: ['CPATH', 'LIBRARY_PATH']
+
+The configuration above will generate dotkit module files that will not contain
+modifications to either ``CPATH`` or ``LIBRARY_PATH`` and environment module
+files that instead will contain these modifications.
+
+^^^^^^^^^^^^^^^^^^^^^
+Autoload dependencies
+^^^^^^^^^^^^^^^^^^^^^
+
+In some cases it can be useful to have module files directly autoload their dependencies.
+This may be the case for Python extensions, if not activated using ``spack activate``:
+
+.. code-block:: yaml
+
+   modules:
+     tcl:
+       ^python:
+         autoload: 'direct'
+
+The configuration file above will produce module files that will automatically
+load their direct dependencies. The allowed values for the  ``autoload`` statement
+are either ``none``, ``direct`` or ``all``.
+
+.. note::
+  TCL prerequisites
+     In the ``tcl`` section of the configuration file it is possible to use
+     the ``prerequisites`` directive that accepts the same values as ``autoload``.
+     It will produce module files that have a ``prereq``
+     statement instead of automatically loading other modules.
+
+========================
+Module files maintenance
+========================
+
+Spack not only provides great flexibility in the generation of module files
+and in the customization of both their layout and content, but also ships with
+a tool to ease the burden of their maintenance in production environments.
+This tool is the ``spack module`` command:
+
+.. command-output:: spack module --help
+
+------------------------
+``spack module refresh``
+------------------------
+
+The command that regenerates module files to update their content or their layout
+is ``module refresh``:
+
+.. command-output:: spack module refresh --help
+
+A set of packages can be selected using anonymous specs for the optional
+``constraint`` positional argument. The argument ``--module-type`` identifies
+the type of module files to refresh. Optionally the entire tree can be deleted
+before regeneration if the change in layout is radical.
+
+-------------------
+``spack module rm``
+-------------------
+
+If instead what you need is just to delete a few module files, then the right
+command is ``module rm``:
+
+.. command-output:: spack module rm --help
+
+.. note::
+  We care about your module files!
+   Every modification done on modules that are already existing will
+   ask for a confirmation by default. If the command is used in a script it is
+   possible though to pass the ``-y`` argument, that will skip this safety measure.
+
+
+
+.. _extensions:
+
+----------------------
+``spack module loads``
+----------------------
+
```

Comment:
This is not module file maintenance, and should not be here.  It should be up top, next to discussion on `spack load`.